### PR TITLE
[DOCS] Updating options for torch.compile

### PR DESCRIPTION
**Pull request by sgolebiewski-intel**
Replacing environment variables, as per request in the following ticket: 128406

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: docs/articles_en/openvino_workflow/torch_compile.rst [hunk 1]
```diff
@@ -68,20 +68,56 @@ To use ``torch.compile``, you need to add an import statement and define one of
...
-  By default, the OpenVINO IR is saved in the ``cache`` sub-directory, created in the application's root directory. 
+Options
+++++++++++++++++++++
+
+You can use a dictionary with options for ``torch.compile`` to set the backend device, 
+enable model caching, set the cache directory etc. See the example below:
+
+.. code-block:: python
+
+   model = torch.compile(model, backend="openvino", options = {"device" : "CPU", "model_caching" : True})
```
Comment (suryasidd) [inline docs/articles_en/openvino_workflow/torch_compile.rst:80]: Hey Karol,

Can we please add the third option as well here. I forgot to add it. 
{"device" : "CPU", "model_caching" : True, "cache_dir": "./model_cache"}

File: docs/articles_en/openvino_workflow/torch_compile.rst [hunk 2]
```diff
@@ -68,20 +68,56 @@ To use ``torch.compile``, you need to add an import statement and define one of
...
-  this variable to GPU.0, for example, will make the application use the integrated graphics processor instead.
-* **OPENVINO_TORCH_MODEL_CACHING**: enables saving the optimized model files to a hard drive, after the first application run.
-  This makes them available for the following application executions, reducing the first-inference latency.
-  By default, this variable is set to ``False``. Setting it to ``True`` enables caching.
-* **OPENVINO_TORCH_CACHE_DIR**: enables defining a custom directory for the model files (if model caching set to ``True``).
-  By default, the OpenVINO IR is saved in the ``cache`` sub-directory, created in the application's root directory. 
+Options
+++++++++++++++++++++
+
+You can use a dictionary with options for ``torch.compile`` to set the backend device,
```
Comment (suryasidd) [inline docs/articles_en/openvino_workflow/torch_compile.rst:75]: Can we make sure to reflect that these options are optional ? 
Something like, 
You can use an optional dictionary of options for torch.compile to set the backend_device, .... Don't know if it's confusing with word options. :)

File: docs/articles_en/openvino_workflow/torch_compile.rst [hunk 3]
```diff
@@ -68,20 +68,56 @@ To use ``torch.compile``, you need to add an import statement and define one of
...
+  on CPU. If you set this variable to ``GPU.0``, for example, the application will 
+  use the integrated graphics processor instead.
+* ``model_caching`` - enables saving the optimized model files to a hard drive, 
+  after the first application run. This makes them available for the following 
+  application executions, reducing the first-inference latency. By default, this 
+  variable is set to ``False``. Set it to ``True`` to enable caching.
+* ``cache_dir`` - enables defining a custom directory for the model files (if 
+  ``model_caching`` is set to ``True``). By default, the OpenVINO IR is saved 
+  in the cache sub-directory, created in the application's root directory.
+* Any OpenVINO configuration option. For details, refer to the :ref:`OpenVINO Advanced Features <openvino-advanced-features>`.
```
Comment (suryasidd) [inline docs/articles_en/openvino_workflow/torch_compile.rst:102]: Can we change this to,
``config`` - enables passing any OpenVINO configuration option as a dictionary to this variable. For details about the various options :ref

File: docs/articles_en/openvino_workflow/torch_compile.rst [hunk 4]
```diff
@@ -68,20 +68,56 @@ To use ``torch.compile``, you need to add an import statement and define one of
...
+
+.. important::
+   
+   ``device``, ``model_caching``, and ``cache_dir`` have replaced options that were available in the previous release:
```
Comment (suryasidd) [inline docs/articles_en/openvino_workflow/torch_compile.rst:106]: These are going to replace the environment variables in the coming release Karol. These have not been deprecated yet. 
So we can say, setting options through environment variables will be deprecated from the upcoming release. 
Options will replace environment variables

**Review by suryasidd — CHANGES_REQUESTED** [review]

File: docs/articles_en/openvino_workflow/torch_compile.rst (hunk 1 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/openvino_workflow/torch_compile.rst:80]:
```suggestion
   model = torch.compile(model, backend="openvino", options = {"device" : "CPU", "model_caching" : True, "cache_dir": "./model_cache"})
```

File: docs/articles_en/openvino_workflow/torch_compile.rst (hunk 2 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/openvino_workflow/torch_compile.rst:75]:
```suggestion
You can use an optional dictionary of options for ``torch.compile`` to set the backend device, 
```

File: docs/articles_en/openvino_workflow/torch_compile.rst (hunk 3 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/openvino_workflow/torch_compile.rst:102]:
```suggestion
* ``config`` - enables passing any OpenVINO configuration option as a dictionary to this variable. For details on the various options, refer to the :ref:`OpenVINO Advanced Features <openvino-advanced-features>`.
```

File: docs/articles_en/openvino_workflow/torch_compile.rst (hunk 4 above)
Comment (sgolebiewski-intel) [reply docs/articles_en/openvino_workflow/torch_compile.rst:106]:
```suggestion
   Setting options through environment variables will be deprecated from the upcoming release. The ``device``, ``model_caching``, and ``cache_dir`` options will replace the environment variables:
```

Comment (kblaszczak-intel) [reply docs/articles_en/openvino_workflow/torch_compile.rst:106]: I guess the term "in the coming release" makes it a bit ambiguous. The question is: are these new options going to be available in 23.3 for production purposes? 
* If so, we should say that they have replaced the old ones. Even if the old ones are still available (e.g. for backwards compatibility). They are deprecated, and will be removed entirely in future releases. 
* If the environment variables are still the main way to set options for 23.3, then we should change the order of information - first the default, then the way that will be available in future releases...

File: docs/articles_en/openvino_workflow/torch_compile.rst (hunk 2 above)
Comment (kblaszczak-intel) [reply docs/articles_en/openvino_workflow/torch_compile.rst:75]: We'll rephrase it to stress that options are not mandatory and make the "dictionary" less pronounced.

File: docs/articles_en/openvino_workflow/torch_compile.rst [hunk 5]
```diff
@@ -68,20 +68,56 @@ To use ``torch.compile``, you need to add an import statement and define one of
...
-* **OPENVINO_TORCH_MODEL_CACHING**: enables saving the optimized model files to a hard drive, after the first application run.
-  This makes them available for the following application executions, reducing the first-inference latency.
-  By default, this variable is set to ``False``. Setting it to ``True`` enables caching.
-* **OPENVINO_TORCH_CACHE_DIR**: enables defining a custom directory for the model files (if model caching set to ``True``).
-  By default, the OpenVINO IR is saved in the ``cache`` sub-directory, created in the application's root directory. 
+Options
+++++++++++++++++++++
+
+You can use a dictionary with options for ``torch.compile`` to set the backend device, 
+enable model caching, set the cache directory etc. See the example below:
```
Comment (sgolebiewski-intel) [inline docs/articles_en/openvino_workflow/torch_compile.rst:76]:
```suggestion
enable model caching, set the cache directory etc. You can use a dictionary of options such as below:
```

File: docs/articles_en/openvino_workflow/torch_compile.rst (hunk 4 above)
Comment (suryasidd) [reply docs/articles_en/openvino_workflow/torch_compile.rst:106]: No the options will be the main way to set the parameters but the users can use env variables as well. We can say that options replaced the environment variables and the env variables will be entirely removed from future releases

Comment (kblaszczak-intel) [reply docs/articles_en/openvino_workflow/torch_compile.rst:106]: Right, so I suggest putting it like this and including the whole thing in the drop-down:
```suggestion
   The environment variables used in the previous release are still available but are not
   recommended. They will be removed fully in future releases.
   Click to view the deprecated options:
```

**Review by suryasidd — APPROVED** [review]

File: docs/articles_en/openvino_workflow/torch_compile.rst (hunk 4 above)
Comment (suryasidd) [reply docs/articles_en/openvino_workflow/torch_compile.rst:106]: Yep this sounds good Karol.

Comment (sgolebiewski-intel) [reply docs/articles_en/openvino_workflow/torch_compile.rst:106]:
```suggestion
   The environment variables used in the previous release are still available but are not recommended. They will be removed fully in future releases.
```

File: docs/articles_en/openvino_workflow/torch_compile.rst [hunk 6]
```diff
@@ -68,20 +68,56 @@ To use ``torch.compile``, you need to add an import statement and define one of
...
+* ``cache_dir`` - enables defining a custom directory for the model files (if 
+  ``model_caching`` is set to ``True``). By default, the OpenVINO IR is saved 
+  in the cache sub-directory, created in the application's root directory.
+* ``config`` - enables passing any OpenVINO configuration option as a dictionary to this variable. For details on the various options, refer to the :ref:`OpenVINO Advanced Features <openvino-advanced-features>`.
+
+.. important::
+   
+   ``device``, ``model_caching``, and ``cache_dir`` have replaced options that were available in the previous release:
+
+   .. dropdown:: Click to view the previous options.
```
Comment (sgolebiewski-intel) [inline docs/articles_en/openvino_workflow/torch_compile.rst:108]:
```suggestion
   .. dropdown:: Click to view the deprecated options.
```

**Review by kblaszczak-intel — APPROVED** [review]
